### PR TITLE
[device][platform] Update Inventec new platform d6356

### PR DESCRIPTION
**Pull request by linalvin**
* support stretch
    * remove binary files and remove non-supported attributes in config.bcm

**- What I did**
support the new Inventec platform d6356

**- How I did it**
* add platform module
* QoS support
* add d6356 to list in mk files

**- How to verify it**
build the sonic image
**- Description for the changelog**

support new Inventec platform d6356  and support for stretch

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: device/inventec/x86_64-inventec_d6356-r0/INVENTEC-D6356/port_config.ini [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# name          lanes             alias	        index
+Ethernet0       1                 Ethernet0     0
```
Comment (lguohan) [inline device/inventec/x86_64-inventec_d6356-r0/INVENTEC-D6356/port_config.ini]: index must start from 1.

Comment (linalvin) [reply device/inventec/x86_64-inventec_d6356-r0/INVENTEC-D6356/port_config.ini]: according to https://github.com/Azure/SONiC/wiki/Porting-Guide
we can set the index from 0 if our front panel ports are starting from 0

Comment (linalvin) [reply device/inventec/x86_64-inventec_d6356-r0/INVENTEC-D6356/port_config.ini]: in sfputil.py, this platform port mapping is from 0

**Comment by lguohan** [comment]
>     .addr_len_smf      =0x50,  .page_len_smf      =-1,  .offset_len_smf      =14,   .length_len_smf      =1,

why not use the eeprom module in sonic?

---
Refers to: platform/broadcom/sonic-platform-modules-inventec/d6356/modules/transceiver.c:28 in 9353030. [](commit_id = 9353030674904aa32e381032a0f7ce22ba922efc, deletion_comment = False)

File: platform/broadcom/sonic-platform-modules-inventec/d6356/modules/i2c-mux-pca9541.c [hunk 2]
```diff
@@ -0,0 +1,559 @@
+/*
+ * I2C multiplexer driver for PCA9541 bus master selector
```
Comment (lguohan) [inline platform/broadcom/sonic-platform-modules-inventec/d6356/modules/i2c-mux-pca9541.c]: >PCA9541 [](start = 30, length = 7)

why not use default kernel module?

File: platform/broadcom/sonic-platform-modules-inventec/d6356/modules/lpc_ich.c [hunk 3]
```diff
@@ -0,0 +1,1075 @@
+/*
+ *  lpc_ich.c - LPC interface for Intel ICH
```
Comment (lguohan) [inline platform/broadcom/sonic-platform-modules-inventec/d6356/modules/lpc_ich.c]: why not use default kernel module?

File: platform/broadcom/sonic-platform-modules-inventec/d6356/modules/ucd9000.c [hunk 4]
```diff
@@ -0,0 +1,246 @@
+/*
+ * Hardware monitoring driver for UCD90xxx Sequencer and System Health
+ * Controller series
+ *
+ * Copyright (C) 2011 Ericsson AB.
+ *
```
Comment (lguohan) [inline platform/broadcom/sonic-platform-modules-inventec/d6356/modules/ucd9000.c]: why not use kernel default?

File: platform/broadcom/sonic-platform-modules-inventec/d6356/modules/i2c-mux-pca9541.c (hunk 2 above)
Comment (linalvin) [reply platform/broadcom/sonic-platform-modules-inventec/d6356/modules/i2c-mux-pca9541.c]: removed and use dufault

File: platform/broadcom/sonic-platform-modules-inventec/d6356/modules/lpc_ich.c (hunk 3 above)
Comment (linalvin) [reply platform/broadcom/sonic-platform-modules-inventec/d6356/modules/lpc_ich.c]: removed and use default

File: platform/broadcom/sonic-platform-modules-inventec/d6356/modules/ucd9000.c (hunk 4 above)
Comment (linalvin) [reply platform/broadcom/sonic-platform-modules-inventec/d6356/modules/ucd9000.c]: removed and use default

**Comment by lguohan** [comment]
why not use the eeprom module in sonic?

**Comment by linalvin** [comment]
> why not use the eeprom module in sonic?

base on the i2c topology design, we have our own eeprom module call inv_eeprom

File: device/inventec/x86_64-inventec_d6356-r0/minigraph.xml [hunk 5]
```diff
@@ -0,0 +1,144 @@
+<DeviceMiniGraph xmlns="Microsoft.Search.Autopilot.Evolution" xmlns:i="http://www.w3.org/2001/XMLSchema-instance">
```
Comment (lguohan) [inline device/inventec/x86_64-inventec_d6356-r0/minigraph.xml]: please move minigraph.xml, it is no longer needed.

**Pull request merged**